src/lib.rs.in

```diff
@@ -17,6 +17,8 @@ class PublicKey {
     private constructor = empty;
     static_method public_key::validate(bytes: &[i8]) -> Result<PublicKey, String>;
     method public_key::as_bytes(&self) -> Vec<i8>; alias asBytes;
+    method hash(&self) -> i32; alias hashCode;
```

Comment:
This is very clever! Rely on the rust `Hash` impl and then just take the first 32 bits as a i32. Curious, if the type doesn't have a `Hash` instance, was the compiler message reasonable?

Reply:
Yes, it will use the compiler message from Rust that it's not satisfying `<T: Hash>`:

> the trait `std::hash::Hash` is not implemented for `Foo`